transcribe.py

```diff
@@ -1,102 +1,296 @@
 #!/usr/bin/python3
 
 import argparse
-import base64
 import json
 import pathlib
-import sys
 
 import asyncio
-import aiohttp
+import numpy as np
+from openai import AsyncOpenAI
+from openai.types.audio import Transcription
+import pandas as pd
 
 import utils
 
-NUM_ELEMENTS_PER_BATCH = 10
 
-async def fetch(session, url, data):
-    headers = { 'Content-Type' : 'application/json' }
-    async with session.post(url, data=json.dumps(data), headers=headers) as response:
-        if not response.ok:
-            print(response)
+def map_transcription_response_to_result(transcription: Transcription):
+    return {
+        "segments": [
+            {
+                "id": segment["id"],
+                "seek": segment.get("seek") or 0.0,
```

Comment:
Why do we have all those 'or' cases?
When does that happen?

Reply:
I am ashamed to say I have no CUDA device, this makes "faster-whisper" super-slow-whisper for me. Thus, I transcribe using whisper.cpp, which might not have all of those in the response.
To make sure we serialize all properly, I had those safety Or's added so no casting errors during serialization to Parquet will throw. (See my other comment about Parquet usage)

Reply:
No worries, just add a comment :)